homeassistant/components/orvibo/switch.py

```diff
@@ -1,120 +1,170 @@
-
-    add_entities_callback(switches)
+            return
+
+        ir.async_create_issue(
+            hass,
+            DOMAIN,
+            f"eyaml_deprecation_{switch.get('mac').replace(':', '').lower()}",
+            breaks_in_ha_version="2026.11.0",
+            is_fixable=False,
+            is_persistent=True,
+            severity=ir.IssueSeverity.WARNING,
+            translation_key="yaml_deprecation",
+            translation_placeholders={
+                "host": switch.get("host"),
+                "mac": switch.get("mac"),
+                "name": switch.get("name"),
+            },
+        )
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: S20ConfigEntry,
+    async_add_entities: AddConfigEntryEntitiesCallback,
+) -> None:
+    """Setup Entry."""
+    switch = []
+    switch.append(
+        S20Switch(entry.data[CONF_NAME], entry.data[CONF_HOST], entry.data[CONF_MAC]),
+    )
+    async_add_entities(switch)
 
 
 class S20Switch(SwitchEntity):
     """Representation of an S20 switch."""
 
-    def __init__(self, name, s20):
+    _attr_has_entity_name = True
+    _attr_name = None
+    _attr_should_poll = True
+
+    def __init__(self, name, host, mac):
         """Initialize the S20 device."""
 
         self._name = name
-        self._s20 = s20
+        self._host = host
+        self._mac = mac
         self._state = False
         self._exc = S20Exception
-
-    @property
-    def name(self):
-        """Return the name of the switch."""
-        return self._name
+        self._s20 = S20(self._host, self._mac)
+        self._unique_id = "S20Switch_" + self._mac
 
     @property
     def is_on(self):
         """Return true if device is on."""
         return self._state
 
-    def update(self) -> None:
-        """Update device state."""
-        try:
-            self._state = self._s20.on
-        except self._exc:
-            _LOGGER.exception("Error while fetching S20 state")
+    @property
+    def unique_id(self):
+        """Return the unique id."""
+        return self._unique_id
+
+    @property
+    def device_info(self) -> DeviceInfo:
+        """Return the device info."""
+        return DeviceInfo(
+            identifiers={
+                # Serial numbers are unique identifiers within a specific domain
+                (DOMAIN, self._unique_id)
+            },
+            name=self._name,
+            manufacturer="Orvibo",
+            model="S20",
+        )
 
     def turn_on(self, **kwargs: Any) -> None:
         """Turn the device on."""
         try:
             self._s20.on = True
-        except self._exc:
-            _LOGGER.exception("Error while turning on S20")
+        except self._exc as err:
+            raise HomeAssistantError(
+                translation_domain=DOMAIN,
+                translation_key="turn_on_error",
+                translation_placeholders={"name": self._name},
+            ) from err
```

Comment:
I like these changes, but let's keep them for a followup so we can focus on the config flow here